micro_sam/util.py

```diff
@@ -203,7 +156,7 @@ def get_sam_model(
     Returns:
         The segment anything predictor.
     """
-    checkpoint = _get_checkpoint(model_type, checkpoint_path)
+    checkpoint = MODELS.fetch(model_type)
```

Comment:
We can just add something like this to support `checkpoint_path`:
```python
if checkpoint_path is None:
  checkpoint = MODELS.fetch(model_type)
else:
  checkpoint = checkpoint_path
```

Reply:
So I don't see any problem with mixing up hashes.

Reply:
I'm confused - if the weights are different, the hash value for the object should be different too, right? So how can anybody possibly do some extra fine tuning on a particular model, save the new weights, and then use checkpoint_path and give it the hash value stored in the pooch registry?

I might misunderstand what is actually happening with `checkpoint_path`.

Reply:
Sorry I should have written a better description here. I will try to explain what happens if a `checkpoint_path` is passed or not:
- if `checkpoint_path` is given: this is a local path to sam weigths. E.g. if a user has fine-tuned on their own data. In this case we don't download any weights but initialize the model from these local weights. We do not check the weight file against any hashes.
- if `checkpoint_path` is not given: we download the SAM weights corresponding to `model_type`. (I agree that this name is confusing, see comment below). We can check their hash.

> So how can anybody possibly do some extra fine tuning on a particular model, save the new weights, and then use checkpoint_path and give it the hash value stored in the pooch registry?

When finetuning we don't check the weights against the hash values in the pooch registry.